tests/broken_options_test.cpp

```diff
@@ -243,6 +543,31 @@ about
   -u [ --togg ]                           some other toggle
 )EXPECTED");
     }
+
+    SECTION("Multiple groups with same name should be one group")
+    {
+        nitro::broken_options::parser parser("app_name", "about");
+
+        std::stringstream s;
+
+        auto grp1 = parser.group("group1");
+        auto grp2 = parser.group("group2");
+
+        grp1.toggle("tog", "should work");
+        REQUIRE_THROWS_AS(grp2.toggle("tog", "should fail"), nitro::broken_options::parser_error);
+    }
+
+    SECTION("Groups work 2")
+    {
+        nitro::broken_options::parser parser("app_name", "about");
+
+        std::stringstream s;
+
+        auto grp1 = parser.group("group1");
+        auto grp2 = parser.group("group1");
+
+        REQUIRE(grp1 == grp2);
```

Comment:
This condition looks okay, but what about (pseudo-code):

```
auto opt1 = grp1.option("opt");
auto opt2 = grp2.option("opt");

REQUIRE(opt1 == opt2);
```

Reply:
`auto opt2 = grp2.option("opt");` raises an exception because 'opt' is already defined in grp1

Reply:
This works well, if you use auto& grpX ...
I also added a test for this case.